src/util/moveThought.js

```diff
@@ -5,12 +5,17 @@ import { timestamp } from './timestamp.js'
 /** Returns a new thought that has been moved either between contexts or within a context (i.e. changed rank) */
 export const moveThought = (thought, oldContext, newContext, oldRank, newRank) => {
   if (typeof thought === 'string') throw new Error('removeContext expects an [object] thought, not a [string] value.')
+
+  //check if both of two context exist in thought
+  const bothExists = [oldContext, newContext]
+    .every(context => (thought.contexts || []).find(parent => equalArrays(parent.context, context) ? true : false ));
```

Comment:
I think you only have to check against `newContext` here. We already know that `thought` is in `oldContext`.

Also you can omit ` ? true : false` as it is redundant.